spec/lib/declare_schema/migration_generator_spec.rb

```diff
@@ -860,21 +862,39 @@ class Creative < ActiveRecord::Base
       end
 
       expect(Generators::DeclareSchema::Migration::Migrator.run).to(
-        migrate_up(<<~EOS.strip)
-          create_table :advertisers, id: :bigint#{create_table_charset_and_collation} do |t|
-            t.string :name, limit: 250, null: false#{charset_and_collation}
-          end
-          create_table :advertisers_creatives, primary_key: [:advertiser_id, :creative_id]#{create_table_charset_and_collation} do |t|
-            t.integer :advertiser_id, limit: 8, null: false
-            t.integer :creative_id, limit: 8, null: false
-          end
-          create_table :creatives, id: :bigint#{create_table_charset_and_collation} do |t|
-            t.string :url, limit: 500, null: false#{charset_and_collation}
-          end
-          add_index :advertisers_creatives, [:creative_id], name: :index_advertisers_creatives_on_creative_id
-          add_foreign_key :advertisers_creatives, :advertisers, column: :advertiser_id, name: :advertisers_creatives_FK1
-          add_foreign_key :advertisers_creatives, :creatives, column: :creative_id, name: :advertisers_creatives_FK2
-      EOS
+        if ActiveSupport.version >= Gem::Version.new('7.0.0') && Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.1.0')
+          migrate_up(<<~EOS.strip)
+            create_table :advertisers_creatives, primary_key: [:advertiser_id, :creative_id]#{create_table_charset_and_collation} do |t|
+              t.integer :advertiser_id, limit: 8, null: false
+              t.integer :creative_id, limit: 8, null: false
+            end
+            create_table :creatives, id: :bigint#{create_table_charset_and_collation} do |t|
+              t.string :url, limit: 500, null: false#{charset_and_collation}
+            end
+            create_table :advertisers, id: :bigint#{create_table_charset_and_collation} do |t|
+              t.string :name, limit: 250, null: false#{charset_and_collation}
+            end
```

Comment:
📓 This only difference on this block and the below block is that the create table ordering is different, seems that rails 7+ on ruby 3.1+ orders by most recent declared models first.